test/cljc/reitit/ring_coercion_test.cljc

```diff
@@ -680,6 +680,77 @@
                      (call (request "application/json" "application/transit" {:request :json :response :json}))))))))))))
 
 
+#?(:clj
+   (deftest response-coercion-test
+     (doseq [[coercion schema-200 schema-default]
+             [[malli/coercion
+               [:map [:a :int]]
+               [:map [:b :int]]]
+              [schema/coercion
+               {:a s/Int}
+               {:b s/Int}]
+              [spec/coercion
+               {:a int?}
+               {:b int?}]]]
+       (testing (str coercion)
+         (let [app (ring/ring-handler
+                    (ring/router
+                     ["/foo" {:post {:responses {200 {:content {:default {:schema schema-200}}}
+                                                 201 {:content {"application/edn" {:schema schema-200}}}
+                                                 202 {:description "status code explicitly mentioned, but no :content map"}
```

Comment:
The interaction here between `:content` and `:default` will probably mislead users.

- If a status code is listed, but has no `:content`, the coercion for the `:default` status will apply
- If a response code has a `:content`, it implicitly means non-mentioned content-types will have _no coercion at all_

It would be clearer, IMO, to have the `:default` response only apply when the status code isn't listed. That is, in this case, status 202 should have no response coercion at all, for any content type.

Reply:
Yeah, I agree that the `:default` response should only apply if there is an unlisted status code.

As in #691 , it would be good to explicitly and separately specify and document the recommended way of specifying that an empty response body is expected.